handlers/status_handler.cc

```diff
@@ -33,15 +34,17 @@ RequestHandler::Status StatusHandler::InitStatus(ServerStatus* status) {
   if (status == nullptr) {
     return RequestHandler::MISCONFIGURED_HANDLER;
   }
+  status->GetStatus().stsMx->lock();
   status_ = status;
```

Comment:
This call on the copy constructor is not safe -- you're copying the mutex, for example